sphinx.go

```diff
@@ -131,7 +133,12 @@ func generateSharedSecrets(paymentPath []*btcec.PublicKey,
 	// Within the loop each new triplet will be computed recursively based
 	// off of the blinding factor of the last hop.
 	lastEphemeralPubKey := sessionKey.PubKey()
-	hopSharedSecrets[0] = generateSharedSecret(paymentPath[0], sessionKey)
+	hopSharedSecrets[0], err = generateSharedSecret(
```

Comment:
No other errors in this function as is atm, so perhaps we should remove the error declaration above and just assign+declare the variable here? (walrus operator: `:=`)